lib/rules/prop-types.js

```diff
@@ -164,6 +164,16 @@ module.exports = function(context) {
     return tokens.length && tokens[0].value === '...';
   }
 
+  function getKeyValue(node) {
+    var key = node.key;
+    if (key) {
+      if (key.type === 'Identifier') {
+        return key.name;
+      }
+      return key.value;
+    }
```

Comment:
Yep, there is really a lot of different ways to declare/use propTypes, it become hard to check all of them.